features/steps/utils.py

```diff
@@ -88,6 +88,28 @@ def get_resource(context, resource):
     return response
 
 
+def make_request(cache, url, headers):
+    if url in cache:
+        return cache[url]
+
+    with requests.get(url, headers=headers, stream=True) as response:
+        # Limit response size
+        # First check the content-length header, if present. This will prevent downloading overly large
+        # responses, assuming a well configured server.
+        content_length = int(response.headers.get('content-length', 0))
+        assert content_length < CONTENT_LENGTH_LIMIT, \
+                ERROR_CONTENT_LENGTH.format(content_length, CONTENT_LENGTH_LIMIT)
+
+        # Now, complete the download, and check the response size.
+        content_length = len(response.content)
```

Comment:
This property access on `response.content` is what triggers the actual synchronous read/download of the entire body? This is what makes me love/hate Python :-)

Reply:
It's actually a pretty sane API http://docs.python-requests.org/en/master/user/advanced/#body-content-workflow, all things considered. If you want an asynchronous stream, use the `iter_content` property. If you want to just download the body as normal, use the `content` property. And this all only gets weird when `stream=True`.

Reply:
Sure -- it's just the notion of a property access (rather than a function call) triggering a blocking step that surprises me.